openapi.yaml

```diff
@@ -55,8 +55,485 @@ paths:
+            and following characters can be 'a'-'z' or 'A'-'Z' or '0'-'9'
+        required: true
+        schema:
+          type: string
+      - name: propertyName
+        in: path
+        description: |
+          Property Name. 
+          The first character of the namespace must be in the 'a'-'z' or 'A'-'Z' ranges,
+          and following characters can be 'a'-'z', 'A'-'Z', '0'-'9', '.' (period), '-' (dash) or '_' (underscore)
+        required: true
+        schema:
+          type: string
+      responses:
+        '200':
+          description: Action Message
+          content:
+            application/json:
+              schema:
+                type: string
+              example: Successfully deleted property propertyName in namespace
+        '401':
+          description: Unauthorized as valid authentication has not been provided
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/APIError'
+              examples:
+                unauthorizederror:
+                  value:
+                    error_code: 5401
+                    error_message: "GAL5401E: Unauthorized. Please ensure you have provided a valid 'Authorization' header with a valid bearer token and try again."
+                  summary: The request was unauthenticated so is unauthorized.
+        '404':
+          description: Bad Request
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/APIError'
+              examples:
+                namespaceerror: 
+                  value: 
+                    error_code: 5017
+                    error_message: "GAL5017E: Error occured when trying to access namespace 'abc!'. The Namespace provided is invalid."
+                  summary: An error occured when trying to retrieve the CPS namespaces
+                newpropertyerror:
+                  value:
+                    error_code: 5018
+                    error_message: "GAL5018E: Error occured when trying to access property 'propertyName'. The property name provided is invalid."
+                  summary: An error occured because the property does not exist
+        '500':
+          description: Error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/JsonError'
+              examples:
+                genericerror:
+                  value:
+                    error_code: 5000
+                    error_message: "GAL5000E: Error occured when trying to access the endpoint. Report the problem to your Galasa Ecosystem owner."
+                  summary: An error occured when trying access the endpoint
+
+  #--------------------------------------
+  # Deprecated CPS API endpoints
+  #--------------------------------------
   /cps/namespace:
     get:
+      deprecated: true
```

Comment:
When we deprecate something, we should say which version it was deprecated in, and that it will be removed at some point in the future. And what the user should be using instead.